src/assignmentfifteen/TaskThree/Program.cs

```
@@ -0,0 +1,88 @@
+                Console.WriteLine($"\nHi I'm task one.I completed my execution on {DateTime.Now}");
+                Console.WriteLine($"\nUNSORTED ARRAY : {string.Join(", ", arrayToSort)}");
+            });
+
+            Thread threadTwo = new (() =>
+            {
+                threadOne.Join(); // comment this to observe how threads behave in exception.
+                sortedArray = SortArrayInDescendingOrder(arrayToSort);
+                Console.WriteLine($"\nHi I'm task Two.I completed my execution on {DateTime.Now}");
+            });
+
+            threadOne.Start();
+            threadTwo.Start();
+            threadTwo.Join(); // comment this to observe the behavior of main thread.
+
+            try
+            {
+                Console.WriteLine($"\nSORTED ARRAY   : {string.Join(", ", sortedArray)}");
+            }
+            catch (Exception)
+            {
+                Utility.SetAlertTextColor($"Exception caught.", "Red");
+            }
+
+            Console.WriteLine($"\nBye!!! Exiting Program on {DateTime.Now}");
+        }
+
+        /// <summary>
+        /// Generates the array of integers.
+        /// </summary>
+        /// <returns>The integer array.</returns>
+        public static int[] GenerateArrayOfIntegers()
+        {
+            int[] array = new int[100];
+            for (int i = 1; i < 100; i++)
+            {
+                array[i] = i;
+            }
+
+            return array;
+        }
+
+        /// <summary>
+        /// Sorts the given integer array in descending order.
+        /// </summary>
+        /// <param name="arrayToSort">Array to be sorted.</param>
+        /// <returns>The sorted array.</returns>
+        /// <exception cref="ArgumentNullException">Throws ArgumentNullException.</exception>
+        public static int[] SortArrayInDescendingOrder(int[] arrayToSort)
+        {
+            if (arrayToSort == null)
+            {
+                throw new ArgumentNullException();
```

I feel this exception is not handled.
When `arrayToSort` is null the program breaks. Exception may happen in line number 29.